### PR TITLE
Hide calculate shipping cost when all items in cart do not require shipping

### LINKED ISSUE #982: Hide shipping calculation in cart when shipping is not required
I think we should hide "CALCULATE SHIPPING COST" section in cart, when products that are currently in cart don't require shipping. It should be only displayed when there is at least one item that does require shipping.

### PR DESCRIPTION
**Pull request by szewczykmira**
Closes #982

### CODE REVIEW

**Comment by szewczykmira** [comment]
Preview:
without_shipping
with_shipping

**Comment by patrys** [comment]
I think it would make the most sense to show:

```
---------------------------------------------------------
SUBTOTAL                                           $50.00
SHIPPING: [ Poland  |…]                       from $21.50
---------------------------------------------------------
TOTAL                                         from $71.50
```

Subtotal does not include shipping.

"Delivery calculated at checkout" is misleading as we do show an estimate.

I think the country selection widget would fit better next to its label rather than under "quantity".

**Comment by weronka** [comment]
small style fixes

screenshot 5

File: saleor/cart/utils.py [hunk 1]
```diff
@@ -212,3 +214,27 @@ def func(request, *args, **kwargs):
...
+    cart_total = None
+    local_cart_total = None
+    shipping_required = False
+    total_with_shipment = None
+    local_total_with_shipment = None
+    if cart:
+        cart_total = cart.get_total(discounts=discounts)
+        local_cart_total = to_local_currency(cart_total, currency)
+        shipping_required = cart.is_shipping_required()
+        total_with_shipment = PriceRange(cart_total)
```
Comment (patrys) [inline saleor/cart/utils.py]: I'd change `shipment` to `shipping` for consistency.

File: saleor/static/js/storefront.js [hunk 2]
```diff
@@ -196,11 +196,10 @@ $('.product-form button').click((e) => {
...
 let $deliveryForm = $('.deliveryform');
 let crsfToken = $deliveryForm.data('crsf');
-let $countrySelect = $('#id_country');
-let $newMethod = $('.cart__delivery-info__method');
-let $newPrice = $('.cart__delivery-info__price');
-$countrySelect.on('change', (e) => {
-  let newCountry = $countrySelect.val();
+let countrySelect = '#id_country';
+let $cartSubtotal = $('.cart__subtotal');
+let $deliveryAjax = (e) => {
```
Comment (patrys) [inline saleor/static/js/storefront.js]: Why the dollar sign? It's usually used to indicate jQuery objects.

**Pull request merged**